### PR TITLE
Fix Cash App Payments in live mode

### LINKED ISSUE #3365: Cash App payments fail in live mode
**Describe the bug**
When a customer attempts to place an order with the Cash App payment method selected, the QR code modal does not appear if the Stripe for WooCommerce extension is configured to accept live payments. Instead, the following error is displayed:

```
The payment failed.
```

When assessing the logs in Stripe, this the result of a `payment_intent_payment_attempt_failed` error and an `invalid_request_error` error in request to confirm a PaymentIntent.

**To Reproduce**
1. Enable the Cash App payment method.
2. Ensure that test mode is disabled.
3. Add a product to your cart.
4. Proceed to checkout.
5. Select the Cash App payment method.
6. Place the order.
7. Note the error.
8. Check the logs.
9. Check the network request for the `confirm` request.

**Expected behavior**
The Cash App payment method displays a QR code modal without encountering an error.

**Screenshots**
image

**Additional context**
8590733-zen
8575609-zen
8459070-zen
8394685-zen

### PR DESCRIPTION
**Pull request by james-allan**
Fixes #3365 

## Changes proposed in this Pull Request:

Cash App payments in live mode are currently failing with a generic failure (`Payment failed`). It wasn't immediately obvious why these payments were failing, however @csmcneill theorised that it could be missing payment method data in `confirmCashappPayment()` request based on [this example](https://docs.stripe.com/payments/cash-app-pay/accept-a-payment?web-or-mobile=web&payments-ui-type=direct-api#handle-redirect). 

After discussing this with Stripe, they suggested that our approach of confirming cash App payments on creation and then confirming via JS (`confirmCashappPayment()`) was leading to this error. 

This PR updates our payment intent and setup intent flows for Cash App to make sure we don't confirm on creation - this is similar to voucher PM. This change then had flow on effects like not being able to pass a `return_url` and `mandates` when not confirming immediately. 

<p align="center">
  
  
</p>

So to summerise the changes:

- **Cash App payments and setup intents no longer confirm immediately on creation.** This is following the advice of Stripe reps here: p1725041202438109/1723789646.813549-slack-C7U3Y3VMY and p1724352624625259/1724202287.961919-slack-C9976E5MJ. 
   - They are still confirmed via JS via [`confirmCashappPayment()`](https://docs.stripe.com/js/payment_intents/confirm_cashapp_payment)
- Following that change, **Cash App payments and intents no longer provide a return URL or mandate information** in the create request as its no longer needed and Stripe don't allow passing it in this scenario. 

## Testing instructions

1. Enable the **Cash App** payment method.
1. Ensure that test mode is disabled.
1. Add a product to your cart.
1. Proceed to checkout.
1. Select the Cash App payment method.
1. Place the order.
    - On develop you will see a generic failure displayed on checkout. 
    - On this branch, the cash app modal should appear.

| Develop | This branch |
|--------|--------|
| Screenshot 2024-08-22 at 10 10 47 AM | Screenshot 2024-08-22 at 10 14 35 AM | 

---

-   [ ] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [ ] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)
-   [ ] Added the [needs docs label](https://github.com/woocommerce/woocommerce-gateway-stripe/labels?q=docs) (or does not apply)
-   [ ] Included this PR in the Release Thread scope (or does not apply)

### CODE REVIEW

**Review by wjrosa — COMMENTED** [review]
Not in the context of this PR, but I wonder if we should replicate the same fix for `confirmWechatPayPayment`, [since it also accepts a `payment_method` parameter](https://docs.stripe.com/js/payment_intents/confirm_wechat_pay_payment#stripe_confirm_wechat_pay_payment-data-payment_method).  Not aware of WeChat failures though

**Comment by james-allan** [comment]
I don't think this is the right approach. It fixes the issue where the modal wasn't showing and we we're getting a generic failure, but it causes Stripe to create a new payment method (`pm_xxx`) and then for some reason all these GET PI logs to occur. 

Screenshot 2024-08-27 at 12 07 47 PM

These might be harmless, however, I don't think recreating the payment method.

**Comment by james-allan** [comment]
We're working with Stripe (internal link: p1724202287961919-slack-C9976E5MJ) to work out why confirming the payment via `confirmCashappPayment()` or even `confirmPayment()` results in a payment failure, rather than displaying the Cash App QR code.

The rep from Stripe has confirmed we're using the deferred intent integration, but with client side confirmation - https://docs.stripe.com/payments/accept-a-payment-deferred. She mentioned that their logs indicate the payment is failing at the partner level. 

From what I can tell, it appears this issue is being caused by our approach of creating the payment method via the Stripe payment elements, creating the payment intent using that payment method, and then confirming it client side via [`confirmCashappPayment()`](https://docs.stripe.com/js/payment_intents/confirm_cashapp_payment) or [`confirmPayment()`](https://docs.stripe.com/js/payment_intents/confirm_payment).

The changes currently proposed on this PR of passing `payment_method: type: 'cashapp',` theoretically fixes the issue because it recreates a new `pm_xxx`. We don't actually want to do that because of how we save payment tokens. We would have saved the original PM created via the elements, not a new one that Stripe creates if pass those params.

File: includes/class-wc-stripe-intent-controller.php [hunk 1]
```diff
@@ -1051,7 +1057,7 @@ public function create_and_confirm_setup_intent_ajax() {
 			wp_send_json_error(
 				[
 					'error' => [
-						'message' => $e->getLocalizesdMessage(),
```
Comment (james-allan) [inline includes/class-wc-stripe-intent-controller.php:1054]: There's a typo in this function leading to this error:

```
PHP Fatal error:  Uncaught Error: Call to undefined method WC_Stripe_Exception::getLocalizesdMessage() in wp-content/plugins/woocommerce-gateway-stripe/includes/class-wc-stripe-intent-controller.php:1050
Stack trace:
#0 wp-includes/class-wp-hook.php(324): WC_Stripe_Intent_Controller->create_and_confirm_setup_intent_ajax('')
#1 wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array)
#2 wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#3 wp-content/plugins/woocommerce/includes/class-wc-ajax.php(96): do_action('wc_ajax_wc_stri...')
#4 wp-includes/class-wp-hook.php(324): WC_AJAX::do_wc_ajax('')
#5 wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(false, Array)
#6 wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#7 wp-includes/template-loader.php(13): do_action('template_redire...')
#8 wp-blog-header.php(19): require_once('/Users/jamesall...')
#9 index.php(17): require('/Users/jamesall...')
#10 {main}
```

File: includes/class-wc-stripe-intent-controller.php [hunk 2]
```diff
@@ -1029,7 +1035,7 @@ public function create_and_confirm_setup_intent_ajax() {
 
 			$setup_intent = $this->create_and_confirm_setup_intent( $payment_information );
 
-			if ( empty( $setup_intent->status ) || ! in_array( $setup_intent->status, [ 'succeeded', 'processing', 'requires_action' ], true ) ) {
+			if ( empty( $setup_intent->status ) || ! in_array( $setup_intent->status, [ 'succeeded', 'processing', 'requires_action', 'requires_confirmation' ], true ) ) {
```
Comment (james-allan) [inline includes/class-wc-stripe-intent-controller.php:1039]: Because we're no longer confirming Cash App setup intents immediately we need to add `requires_confirmation` as a valid status we're expecting.

**Comment by james-allan** [comment]
I opening this for review now. I've tested the following scenarios:

- [x] Purchase using Cash App
- [x] Purchase using saved Cash App token
- [x] Adding Cash App token via checkout
- [x] Add Cash App token via My Account > Payment methods

In all the scenarios listed above, payments and setup-intents all work as expected.

While in Live mode I confirmed that the QR code modal is shown on both the checkout and my Account > Payment methods page. _This is where it was previously failing with the "Payment failed"_

<p align="center">
Screenshot 2024-09-02 at 1 45 41 PM
</p> 

---

@wjrosa I've re-requested your review, however I recall you were having issues in the past with Cash App and getting the modal to show so let me know if you'd prefer I redirect this to someone else for review :)

**Comment by wjrosa** [comment]
> I opening this for review now. I've tested the following scenarios:
> 
> * [x]  Purchase using Cash App
> * [x]  Purchase using saved Cash App token
> * [x]  Adding Cash App token via checkout
> * [x]  Add Cash App token via My Account > Payment methods
> 
> In all the scenarios listed above, payments and setup-intents all work as expected.
> 
> While in Live mode I confirmed that the QR code modal is shown on both the checkout and my Account > Payment methods page. _This is where it was previously failing with the "Payment failed"_
> 
> Screenshot 2024-09-02 at 1 45 41 PM
> 
> @wjrosa I've re-requested your review, however I recall you were having issues in the past with Cash App and getting the modal to show so let me know if you'd prefer I redirect this to someone else for review :)

I believe I fixed those modal issues. It is been a while since it happened. But I tried to test this PR, and I couldn't make the Cash App option appear on my checkout with live mode enabled. Maybe that's because my account is only in test mode? (As I don't have valid US documents to move it to live).

I will approve this only by looking at the code (it is good! And the worst that can happen is this not having any effect), but it would be nice to have someone testing this properly 👍 .

**Review by wjrosa — APPROVED** [review]

**Comment by wjrosa** [comment]
Thanks to @csmcneill , I was able to test it now! Working as expected:

Screenshot 2024-09-05 at 17 28 37
